bundle/Command/ScheduledVisibilityUpdateCommand.php

```diff
@@ -0,0 +1,251 @@
+
+            return Command::FAILURE;
+        }
+
+        $allContentTypes = $this->configurationService->isAllContentTypes();
+        $allowedContentTypes = $this->configurationService->getAllowedContentTypes();
+
+        if (!$allContentTypes && count($allowedContentTypes) === 0) {
+            $this->style->warning('No content types configured for scheduled visibility mechanism.');
+
+            return Command::FAILURE;
+        }
+
+        $pager = $this->getPager($allContentTypes, $allowedContentTypes);
+
+        if ($pager->getNbResults() === 0) {
+            $this->style->info('No content found');
+
+            return Command::FAILURE;
+        }
+
+        $limit = $input->getOption('limit');
+        $offset = 0;
+
+        $progressBar = $this->style->createProgressBar($pager->getNbResults());
+        $progressBar->setFormat('debug');
+        $progressBar->start();
+
+        $results = $pager->getAdapter()->getSlice($offset, $limit);
+        while (count($results) > 0) {
+            $this->processResults($results, $progressBar);
+            $offset += $limit;
+            $results = $pager->getAdapter()->getSlice($offset, $limit);
+        }
+
+        $progressBar->finish();
+
+        $this->style->info('Done.');
+
+        return Command::SUCCESS;
+    }
+
+    private function processResults(array $results, ProgressBar $progressBar): void
+    {
+        foreach ($results as $result) {
+            try {
+                $languageId = $result['initial_language_id'];
+                $language = $this->loadLanguage($languageId);
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Language with id #%d does not exist: %s',
+                        $languageId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                $progressBar->advance();
+
+                continue;
+            }
+
+            try {
+                $contentId = $result['id'];
+
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $this->repository->sudo(
+                    fn () => $this->repository->getContentService()->loadContent(
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                $progressBar->advance();
+
+                continue;
+            }
+
+            try {
+                $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
```

Comment:
`$action` is not used now.

Reply:
fixed in 7e232170b985c7e092e8f2e12507adfb8913a8c6